libexec/rtld-elf/rtld.c

```diff
@@ -2552,12 +2569,22 @@ parse_rtld_phdr(Obj_Entry *obj)
 	const Elf_Phdr *ph;
 	Elf_Note *note_start, *note_end;
 
+#ifndef __CHERI_PURE_CAPABILITY__
 	obj->stack_flags = PF_X | PF_R | PF_W;
+#endif
 	for (ph = obj->phdr;  (const char *)ph < (const char *)obj->phdr +
 	    obj->phsize; ph++) {
 		switch (ph->p_type) {
 		case PT_GNU_STACK:
+#ifdef __CHERI_PURE_CAPABILITY__
+			if ((ph->p_flags & PF_X) != 0) {
```

Comment:
I see the following macros that could be used to simplify this.
```
#define rtld_fatal(args...)	do { _rtld_error(args); rtld_die(); } while (0)
#define rtld_require(cond, args...) if (!(cond)) { rtld_fatal(args); }
```

Reply:
Thanks, I went with `rtld_fatal` since that is the common pattern in rtld.c, and it keeps the condition the same as the other clauses that return an error (vs having to invert it).